src/analysis/transcendental/liouville.lean

```diff
@@ -0,0 +1,529 @@
+              repeat { apply fpow_pos_of_pos, norm_cast, assumption <|> linarith }
+            end
+        ... < (2 : ℝ) ^ (n - 1 : ℤ) / 2 ^ (n - 1 : ℤ) :
+            begin
+              rw [show (2 : ℝ) ^ (n - 1 : ℤ) = 2 ^ (n - 1 : ℕ),
+                  by simp only [nat.add_succ_sub_one, add_zero, int.coe_nat_succ,
+                      add_sub_cancel, fpow_coe_nat], div_lt_div_iff],
+              norm_cast,
+              refine mul_lt_mul h_b_ineq (le_refl _) _ _,
+              repeat { apply pow_pos <|> apply pow_nonneg, linarith }
+            end
+        ... = 1 : by { rw div_self, apply fpow_ne_zero_of_ne_zero, linarith } ) },
+    repeat { norm_cast, assumption <|> linarith }}
+end
+
+theorem transcendental_of_is_liouville {x : ℝ} (liouville_x : is_liouville x) :
+  is_transcendental ℤ x :=
+begin
+  have h_irr_x : irrational x := irrational_of_is_liouville liouville_x,
+  rintros ⟨f : polynomial ℤ, h_f_nonzero, h_root⟩,
+  have h_f_deg : 1 < f.nat_degree :=
+    one_lt_nat_degree_of_irrational_root x f h_irr_x h_f_nonzero h_root,
+  replace h_root : (f.map (algebra_map ℤ ℝ)).eval x = 0, { rw aeval_def at h_root, rwa [eval_map] },
+  obtain ⟨A, hA, h⟩ := exists_pos_real_of_irrational_root x h_irr_x f h_f_deg h_root,
+  have h_pow_big_enough : ∃ (r : ℕ), 1 / A ≤ 2 ^ r,
+  { rcases pow_unbounded_of_one_lt (1 / A) (lt_add_one 1) with ⟨n, hn⟩,
+    exact ⟨n, le_of_lt hn⟩ },
+  obtain ⟨r, hr⟩ := h_pow_big_enough,
+  have hr' : 1 / (2 ^ r) ≤ A,
+  { rw [div_le_iff, mul_comm, ←div_le_iff], repeat { assumption <|> apply pow_pos <|> linarith }},
+  obtain ⟨a, b, h_b_gt_1, h_abs_sub_pos, h_abs_sub_small⟩ := liouville_x (r + f.nat_degree),
+  specialize h a b (by linarith),
+  exact lt_irrefl _
+    (calc abs (x - a/b)
+        < 1 / b ^ (r + f.nat_degree) : h_abs_sub_small
+    ... = 1 / (b ^ r * b ^ f.nat_degree) : by rw pow_add
+    ... = 1 / b ^ r * (1 / b ^ f.nat_degree) : by rw one_div_mul_one_div
+    ... ≤ 1 / 2 ^ r * (1 / b ^ f.nat_degree) :
+        begin
+          refine mul_le_mul _ (le_refl _) _ _,
+          { rw [div_le_div_iff, one_mul, one_mul],
+            refine pow_le_pow_of_le_left _ _ r,
+            repeat { norm_cast, linarith <|> apply pow_nonneg <|> apply pow_pos }},
+          repeat { rw one_div_nonneg, apply pow_nonneg, norm_cast, linarith }
+        end
+    ... ≤ A * (1 / b ^ f.nat_degree) :
+        begin
+          apply mul_le_mul,
+          repeat { linarith <|> assumption },
+          rw one_div_nonneg, apply pow_nonneg, norm_cast, linarith
+        end
+    ... = A / b ^ f.nat_degree : by rw ←div_eq_mul_one_div
+    ... < abs (x - a/b) : h)
+end
+
+section construction
+
+noncomputable theory
+open_locale nat
+
+local notation `α` := ∑' (i : ℕ), 1 / (10 : ℝ) ^ i!
+local notation `α_first` k `terms` :=  ∑ i in range (k+1), 1 / (10 : ℝ) ^ i!
+local notation `α_terms_after` k := ∑' i, 1 / (10 : ℝ) ^ (i + (k+1))!
+
+lemma summable_inv_pow_fact : summable (λ i, 1 / (10 : ℝ) ^ i!) :=
```

Comment:
Does it make sense to generalize `10` to a parameter, and then move this to a different file?